### ISSUE TITLE
add "web-console" action to NB resource

### ISSUE THREAD

**Issue by cjdcordeiro** [nuvla/api-server#360]
this creates a new and temporary SSH credentials, and SSHs to the NB via a web console

**Comment by cjdcordeiro** [comment]
putting as low priority for now, and reassigning to @0xbase12 .

A possible implementation could follow this workflow:
 1. add a new synchronous NB op to start a live SSH terminal to the NB. This op does:
    a. create a new SSH key
    b. launch a new job onto the NB, that takes that SSH key, adds it to the edge device (very much like the action "add new ssh key") and then establishes a websocket with Nuvla

Once the console is closed, the websocket is destroyed and the job running in the NB terminates.

**Comment by cjdcordeiro** [comment]
at this time, the current PoC is making use of websockets to mimic an SSH connection, where the NB starts the websocket server and the Nuvla UI connects as a client. This has the following drawbacks:
 - the user must be on the same network as the NB
 - the websocket connection must be secured (wss) (otherwise some browsers, like Chrome, will not allow it), which is hard with the current configuration since the websocket server is initiated by the NB
 - the websocket is not a real SSH, so interactive commands (like ping, vim, etc.) will not work out of the box

---

Instead, one of the following improvements is recommended:

### 1) Switch roles on the websocket-based architecture

There are some frameworks ([Sente](https://github.com/ptaoussanis/sente) and [Chord](https://github.com/jarohen/chord)) for implementing a websocket server in Clojure. This would be better, such that when the user asks for a console, the UI asks the api-server for a new websocket server route, with the proper TLS credentials, and then asks the NB, via a job, to connect to this web server, securely. The UI simply waits for the connection to be established.

**Pros**
- works regardless of whether the NB is reachable or not
- simple and no extra machinery
- full control of the connection, since the communication can be closed server side

**Cons**
- not a real terminal, so interactive commands will not work
- still need to manage websocket connection credentials on top of the SSH credentials


### 2) use a real VNC server tool (like Apache Guacamole)

For a real SSH experience, it would be better to install something like Apache Guacamole on the server, and have the UI simply expose an iframe when the user requests for a terminal.

**Pros**
 - Can easily have multiple connections open in parallel
 - real SSH (supports interactive commands)
 - no need for extra layer of software (websockets) with additional credentials handling mechanisms (TLS creds). SSH creds are enough

**Cons**
  - probably only feasible when the NB is reachable (in PUSH mode)
